src/pages/LogIn/components/LoginForm.tsx

```diff
@@ -28,66 +28,76 @@ const LoginFormComponent: React.FC<AllProps> = ({ setAuthToken }: AllProps) => {
   const navigate = useNavigate();
   const location = useLocation();
 
+  enum LoginErrorMessage {
+    invalidPassword = `Password must have at least 8 characters`,
+    invalidUsername = `Username can not be empty`,
+    invalidCredentials = `Invalid Credentials`,
+    serverError = `There was a problem connecting to the server!`,
+  }
+
   async function handleSubmit(
-    event: React.MouseEvent<HTMLButtonElement, MouseEvent>
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent> | React.KeyboardEvent
   ) {
     event.preventDefault();
     const authURL = `${process.env.REACT_APP_CHRIS_UI_AUTH_URL}`;
     let token;
 
-    if (!usernameValue) {
-      setIsValidUsername(false);
-    }
-    if (!passwordValue) {
-      setIsValidPassword(false);
-    } else {
-      setIsValidUsername(true);
-      setIsValidPassword(true);
+    try {
+      token = await ChrisApiClient.getAuthToken(
+        authURL,
+        usernameValue,
+        passwordValue
+      );
+    } catch (error: unknown) {
+      setShowHelperText(true);
 
-      try {
-        token = await ChrisApiClient.getAuthToken(
-          authURL,
-          usernameValue,
-          passwordValue
-        );
-      } catch (error) {
+      if (error && !passwordValue && !usernameValue) {
```

Comment:
I am curious, is there any declarative way to do this kind of validation instead of using procedural programming? I have heard of in native HTML there is [form validation](https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation). I challenge you to give this some thought.

Reply:
Thank you!! Will work on this!

Reply:
I have been working on the Login form some more but I'm curious what you mean by using native HTML to this particular form. I know how HTML works with putting validation directly on the input fields but this form is made with Patternfly/react-core. 

I have added that the submit button is disabled until credentials are entered so as a result that means both fields are required. However I used the hook isLoginButtonDisabled as found in Patternfly together with useMemo for the disabling of the button as seen in my last commit. Please let me know if you want me to change anything to native HTML.